### ISSUE TITLE
[BUG] Could not find an option named "reporter".

### ISSUE THREAD

**Issue by krillefear** [dart-code-checker/dart-code-metrics#674]
* **Dart code metrics version:** 4.10.0
* **Dart sdk version:** 2.15.1

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
include: package:pedantic/analysis_options.yaml

linter:
  rules:
    - camel_case_types
    - avoid_print
    - constant_identifier_names
    - prefer_final_locals
    - prefer_final_in_for_each

analyzer:
  errors:
    todo: ignore
  exclude:
    - example/main.dart
    # needed until crypto packages upgrade
    - lib/src/database/database.g.dart
  plugins:
    # Show code metrics in IDE
    # - dart_code_metrics

dart_code_metrics:
  metrics:
    cyclomatic-complexity: 20
    number-of-arguments: 4
    maximum-nesting-level: 5
    source-lines-of-code: 50
    # This is still unstable, so handle with care
    maintainability-index: 40
  metrics-exclude:
    - test/**
  rules:
    #- newline-before-return
    - no-boolean-literal-compare
    - no-empty-block
    #- prefer-trailing-comma
    - prefer-conditional-expressions
    - no-equal-arguments
    - no-equal-then-else
    - no-magic-number:
        severity: style # too noisy otherwise atm
    - avoid-late-keyword:
        severity: style # too noisy otherwise atm
    - avoid-non-null-assertion:
        severity: style # too noisy otherwise atm
    - avoid-unused-parameters
    - binary-expression-operand-order
  anti-patterns:
    - long-method
    - long-parameter-list

```

</details>

**What did you do? Please include the source code example causing the issue.**

```dart

```

**What did you expect to happen?**
When I run `dart pub global run dart_code_metrics:metrics lib -r gitlab > code-quality-report.json` I expect to have the code quality report readable for GitLab.

**What actually happened?**
I get the error:

```sh
Could not find an option named "reporter".
Usage: metrics <command> [arguments] <directories>

Global options:
-h, --help       Print this usage information.


    --version    Reports the version of this tool.

Available commands:
  analyze              Collect code metrics, rules and anti-patterns violations.
  check-unused-code    Check unused code in *.dart files.
  check-unused-files   Check unused *.dart files.
  check-unused-l10n    Check unused localization in *.dart files.

Run "metrics help <command>" for more information about a command.

```

**Are you willing to submit a pull request to fix this bug?**
If I have the time in the next days but it might take a lot of time to get used to the code base.

**Comment by vmalyi** [comment]
We're running into the same issue on our CI environment when executing `dart pub global run dart_code_metrics:metrics lib -r gitlab > code-quality-report.json`. 

This started to today without any additional actions from our side.

**Comment by incendial** [comment]
As a fallback, you can add `analyze` after the `dart_code_metrics:metrics` part, should work fine. We'll fix this, but note that in 5.0 it will be required to always pass a command name to the cli.

**Comment by dkrutskikh** [comment]
@krillefear thank you for for bug report. I'll fix them asap and publish patch version.

**Comment by dkrutskikh** [comment]
Fixed in 4.10.1 release.

**Comment by vmalyi** [comment]
@incendial, @dkrutskikh, folks, thank you for the blazing-fast fix for this!

I can confirm that 4.10.1 fixes the issue.
